From: Ben Hutchings <bhutchings@solarflare.com>
Subject: ptp: Make PTP_1588_CLOCK select rather than depend on PPS
Git-commit:  42a7ccef0a06519e2ea0be71d7c0f4ed9198509f
Patch-mainline: v3.8-rc1~139^2~516
References: fate#314848
Acked-by: Jiri Bohac <jbohac@suse.cz>

PTP hardware clock drivers that select PTP_1588_CLOCK must currently
also select PPS.  For those drivers that don't, the user must enable
PPS, then enable PTP_1588_CLOCK, then the driver.  Simplify things for
developers and users by putting this selection in one place.

Signed-off-by: Ben Hutchings <bhutchings@solarflare.com>
Signed-off-by: David S. Miller <davem@davemloft.net>

diff --git a/drivers/net/Kconfig b/drivers/net/Kconfig
index 2fd69a0..5c5913b 100644
--- a/drivers/net/Kconfig
+++ b/drivers/net/Kconfig
@@ -2407,7 +2407,6 @@ if PCH_GBE
 config PCH_PTP
 	bool "PCH PTP clock support"
 	default n
-	select PPS
 	select PTP_1588_CLOCK
 	select PTP_1588_CLOCK_PCH
 	---help---
diff --git a/drivers/net/ethernet/intel/Kconfig b/drivers/net/ethernet/intel/Kconfig
index 6af030a..730ac57 100644
--- a/drivers/net/ethernet/intel/Kconfig
+++ b/drivers/net/ethernet/intel/Kconfig
@@ -118,7 +118,6 @@ config IGB_PTP
 	bool "PTP Hardware Clock (PHC)"
 	default n
 	depends on IGB
-	select PPS
 	select PTP_1588_CLOCK
 	---help---
 	  Say Y here if you want to use PTP Hardware Clock (PHC) in the
@@ -220,7 +219,6 @@ config IXGBE_PTP
 	bool "PTP Clock Support"
 	default n
 	depends on IXGBE
-	select PPS
 	select PTP_1588_CLOCK
 	---help---
 	  Say Y here if you want support for 1588 Timestamping with a
diff --git a/drivers/ptp/Kconfig b/drivers/ptp/Kconfig
index 3a2b033..a58fb52 100644
--- a/drivers/ptp/Kconfig
+++ b/drivers/ptp/Kconfig
@@ -4,12 +4,9 @@
 
 menu "PTP clock support"
 
-comment "Enable Device Drivers -> PPS to see the PTP clock options."
-	depends on PPS=n
-
 config PTP_1588_CLOCK
 	tristate "PTP clock support"
-	depends on PPS
+	select PPS
 	help
 	  The IEEE 1588 standard defines a method to precisely
 	  synchronize distributed clocks over Ethernet networks. The
-- 
1.7.11.4

